### ISSUE TITLE
ArangoDB startup time increased tremendously

### ISSUE THREAD

**Issue by ra0o0f** [arangodb/arangodb#1781]
Hi

i created a collection with a hash index(unique & sparse) on a filed, imported 3.5 million documents. everything worked fine until i restart the os and ArangoDB service can't start after i wait for 10min. i killed the ArangoDB process and moved the index defenition file to another location, and this time ArangoDB service started.

i know next release would have the persistent indices, but is this normal that the ArangoDB service cant start even after 10 min?

os: windows 8 64bit
arangodb: 2.8.1
memory: 4gb (arangodb consume around 600mb at the startup)
cpu: cure 2 due 2.1 gh (arangodb consume around 60% at the startup)

i had the same collection and index on the ArangoDB 2.7, i had no problem with the startup time.(as long as i remember)

**Comment by dothebart** [comment]
As you already pointed out, Indexes have to fit into RAM. Can you post the `figures` of the collections in question? 

    arangosh> db.theCollection.figures()

It may also be helpfull if you could send [a memory dump of such an ArangoDB](https://github.com/arangodb/arangodb/blob/devel/README_maintainers.md#coredump-generation) so we could have a look at what its actually doing.

**Comment by ra0o0f** [comment]
i think the hash index fits in the RAM. when the collection is empty and first i create the index then import the data(importing takes about 5 min), index is created. but if i import the data then create the index, the index would not be created even after 15 min.

### before creating hash index

```javascript
{
  "alive" : {
    "count" : 3367913,
    "size" : 484979408
  },
  "dead" : {
    "count" : 0,
    "size" : 0,
    "deletion" : 0
  },
  "datafiles" : {
    "count" : 13,
    "fileSize" : 454901760
  },
  "journals" : {
    "count" : 1,
    "fileSize" : 33554432
  },
  "compactors" : {
    "count" : 0,
    "fileSize" : 0
  },
  "shapefiles" : {
    "count" : 0,
    "fileSize" : 0
  },
  "shapes" : {
    "count" : 2,
    "size" : 536
  },
  "attributes" : {
    "count" : 5,
    "size" : 0
  },
  "indexes" : {
    "count" : 2, // a skip-list index is also created on the collection
    "size" : 493395400
  },
  "lastTick" : "3138410850595",
  "uncollectedLogfileEntries" : 0,
  "documentReferences" : 0,
  "waitingFor" : "-",
  "compactionStatus" : {
    "message" : "compaction not yet started",
    "time" : "2016-03-15T06:11:11Z"
  }
}
```

![figures](https://cloud.githubusercontent.com/assets/2182530/13769642/85c8ffda-ea96-11e5-9ae3-9ae1603222ff.JPG)

![indexes](https://cloud.githubusercontent.com/assets/2182530/13769646/8f28e478-ea96-11e5-8d13-5a003b008126.JPG)

![info](https://cloud.githubusercontent.com/assets/2182530/13769649/93bd3cd2-ea96-11e5-8175-d2475d9e0f8a.JPG)

### core dump 

this is the core dump of the process while creating hash index after 15 min

http://www.arangoclient.net/arangod.rar

the cpu/ram/io usage:

![start-index](https://cloud.githubusercontent.com/assets/2182530/13771600/3f1bb4d4-eaa4-11e5-87d5-ff284e0f8267.JPG)

**Comment by jsteemann** [comment]
Thanks for the details. We've been able to reproduce & fix this issue.
The fix will be included in 2.7.8 and 2.8.6.
